### PR TITLE
Allow use of JS modules (ES6) with docks

### PR DESCRIPTION
**Pull request by nboisteault**
* Funded by 3Liz

### CODE REVIEW

File: lizmap/modules/view/classes/lizmapMapDockItem.class.php [hunk 1]
```diff
@@ -17,10 +17,11 @@ class lizmapMapDockItem
     public $order = 0;
     public $css = '';
     public $js = '';
+    public $jsParams = array();
     public $icon = '';
     public $menuIconClasses = '';
 
-    public function __construct($id, $title, $content, $order = 0, $css = '', $js = '')
+    public function __construct($id, $title, $content, $order = 0, $css = '', $js = '', $jsParams = array())
```
Comment (laurentj) [inline lizmap/modules/view/classes/lizmapMapDockItem.class.php:24]: Documentation on parameters are missing. 

We should indicate, at least:
- `$js` is an URL of a JS file
- `$jsParams` is an array containing additional attributes for the `<script>` tags, like `type` if not `text/javascript`, or `defer` etc...

Comment (laurentj) [reply lizmap/modules/view/classes/lizmapMapDockItem.class.php:24]: - `$css` is an URL of a JS file

File: lizmap/responses/myHtmlMapResponse.class.php [hunk 2]
```diff
@@ -91,6 +91,24 @@ public function __construct()
         $this->addJSCode($generalJSConfig);
     }
 
+    protected function outputJsScriptTag($fileUrl, $scriptParams)
```
Comment (laurentj) [inline lizmap/responses/myHtmlMapResponse.class.php:94]: I just update Jelix into Lizmap, no need any more to redefine this method.

File: CHANGELOG-3.5.md [hunk 3]
```diff
@@ -6,6 +6,7 @@
 
 - Add Romanian localization PlotLy file for the dataviz panel (contribution from @ygorigor)
 - Allow PostgreSQL geography type (contribution from @flobz)
+- Allow use of JS modules (ES6) with docks.
```
Comment (laurentj) [inline CHANGELOG-3.5.md:9]: ", by indicating the `type` attribute of `<script>` to `lizmapMapDockItem`."

**Review by laurentj — CHANGES_REQUESTED** [review]

File: lizmap/modules/view/classes/lizmapMapDockItem.class.php [hunk 4]
```diff
@@ -17,10 +17,11 @@ class lizmapMapDockItem
     public $order = 0;
     public $css = '';
     public $js = '';
+    public $jsParams = array();
     public $icon = '';
```
Comment (nboisteault) [inline lizmap/modules/view/classes/lizmapMapDockItem.class.php:21]: @mdouchin `$icon` and `$menuIconClasses` are not in the constructor. It seems they have been forgotten. Should we add them or remove them as they seem unused till now ?

**Pull request merged**